File: app-flutter/lib/pages/flutter/popover/PopupMenuButton.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class PopupMenuButtonPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    PopupMenuButtonPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}

class _PageState extends State<PopupMenuButtonPage> {
    int value = 2;
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
                title:Config.appBar(widget.title),
                centerTitle:true,
            ),
            body: ListView(
                padding:EdgeInsets.all(10.0),
                children: <Widget>[
                    PopupMenuButton(
                        initialValue: value,
                        itemBuilder: (BuildContext context){
                            return [
                                PopupMenuItem(
                                    value: 0,
                                    child: Text('flutter'),
                                ),
                                PopupMenuItem(
                                    value: 1,
                                    child: Text('react'),
                                ),
                                PopupMenuItem(
                                    value: 2,
                                    child: Text('angular'),
                                )
                            ];
                        },
                        child:  Text('child click', style:TextStyle(height: 2.5)),
                        onSelected: (value){
                            setState(() {
                                value = value;
                            });
                            print(value);
                        },
                        onCanceled: (){
                            print('取消');
                        },
                    )
                ],
            )
        );
    }
}

File: app-react/src/pages/flutter/popover/PopupMenuButton.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'点击弹出式菜单列表',
	keyword:'PopupMenuButton( )',
	type:'PopupMenuButton',
	url:'PopupMenuButtonPage',
	data:[
		{ keyword:'initialValue', type:'T', pro:'默认值，任意数据类型' },
		{ 
			keyword:'itemBuilder', type:'List<PopupMenuEntry<T>> Function(BuildContext) itemBuilder', pro:'自定义菜单列表',
			code:[ {path:'flutter/popover/PopupMenuButton/itemBuilder'} ],
			link:[
				{ title:'PopupMenuItem', url:'/flutter/popover/PopupMenuItem' },
			]
		},
		{ keyword:'tooltip', type:'String', pro:'操作提示' },
		{ keyword:'elevation', type:'double', pro:'阴影' },
		{ keyword:'padding', type:'EdgeInsetsGeometry', pro:'补白', link:'/flutter/style/offset/EdgeInsets' },
		{ keyword:'icon', type:'Icon', pro:'图标', link:'/flutter/icon/Icon' },
		{ keyword:'offset', type:'Offset', pro:'偏移', link:'/flutter/style/offset/Offset' },
		{ keyword:'onSelected', type:'void Function(T)', pro:'选择时执行', code:[ {path:'flutter/event/onSelected'} ], },
		{ keyword:'onCanceled', type:'void Function( )', pro:'取消后执行', code:[ {path:'flutter/event/onCanceled'} ], },
		{ keyword:'child', type:'Widget', pro:'点击的控件，不能为 button'},
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }